Sort Species

danforbes opened the pull request:
Allow species to be sorted. Note that sorting by Pokedex number is broken, as the numbers appear to be sorted as strings. So, 100 comes before 2. @Eric-Carlton - can I tap you in for an assist?
#47

Eric-Carlton commented:
@DanForbes is sorting by Pokedex number not working in the monster views? I could have sworn I tested that last night

danforbes commented:
I think it's working fine in the monster view, but it's broken in the species view and I can't figure out why. I basically just copied-and-pasted the monster sorting implementation, but it seems to be treating the Pokedex numbers as strings and not numbers for some reason.

Eric-Carlton commented:
Hmm, yeah, I'll take a peek tonight

Eric-Carlton commented:
Take a look at line 56 of pokemonRoutes:

`let id = pokemon.pokemon_id.toString();`

I think that's your problem. You're setting up your model with a Pokedex ID of type string, when it needs to be a number for sorting purposes. Try removing that `.toString()` and see if that fixes it.

danforbes commented:
Yeah, but we use that same exact value when we sort monsters by Pokedex number and that works, right?

Eric-Carlton commented:
Nope, the Pokedex number isn't converted to a string for monsters. Lines 83-84:

```
formattedPokemon.push(new Pokemon(
    pokemon.pokemon_id,
...
```

danforbes commented:
Ah, totally right! Thank you!

Eric-Carlton commented:
No problem!

danforbes commented:
@Eric-Carlton: should be all set, including the additional feature recently discussed in #47. Thanks again for the assist!

dfoksma01 committed:
Sort Species

Allow species to be sorted. Expose numbers that represent the number of monsters of a given species that should be transferred and/or acquired.

#47

Eric-Carlton commented:
Looks great, closing #47
